packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -263,14 +276,29 @@ async function askUserForFilepathConfirmation(
 
 export async function resolveFoamTemplateVariables(
   templateText: string,
-  extraVariablesToResolve: Set<string> = new Set()
+  extraVariablesToResolve: Set<string> = new Set(),
+  givenValues: Map<string, string> = new Map()
 ): Promise<[Map<string, string>, string]> {
-  const givenValues = new Map<string, string>();
   const variables = findFoamVariables(templateText.toString()).concat(
     ...extraVariablesToResolve
   );
   const uniqVariables = [...new Set(variables)];
 
+  const missingFoamVariables = [...givenValues.keys()].filter(
+    x => uniqVariables.indexOf(x) < 0
+  );
+  if (missingFoamVariables.includes('FOAM_SELECTED_TEXT')) {
+    missingFoamVariables.forEach(key => {
+      templateText = `${templateText}
+      \${FOAM_SELECTED_TEXT}`;
+      uniqVariables.push('FOAM_SELECTED_TEXT');
+
+      if (!givenValues.get('FOAM_SELECTED_TEXT')) {
+        givenValues.set('FOAM_SELECTED_TEXT', '');
+      }
```

Comment:
minor: I wonder if for better readibility we should separate the "`FOAM_SELECTED_TEXT` defaults to ''" from the "if the template doesn't include `FOAM_SELECTED_TEXT` append it". 
In that perspective, maybe you could just add `FOAM_SELECTED_TEXT` to the `variables` (if it doesn't already exist) right before computing `uniqVariables`?

Reply:
I feel this won't work. Adding it to variables would break the determination if it exists as a FOAM variable. I did extract it from the if-loop. What do you think?